From ee69edc107142e39736c8863e200fc3929d2e47c Mon Sep 17 00:00:00 2001
From: past-due <30942300+past-due@users.noreply.github.com>
Date: Sat, 19 Jul 2025 13:53:44 -0400
Subject: [PATCH] urlrequest_curl: Remove references to
 CURLSSLBACKEND_SECURETRANSPORT

cURL no longer supports this backend. Eliminates Wdeprecated-declarations warning
---
 src/urlrequest_curl.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/urlrequest_curl.cpp b/src/urlrequest_curl.cpp
index b6ac8e1e903..e91b2b1bb4d 100644
--- a/src/urlrequest_curl.cpp
+++ b/src/urlrequest_curl.cpp
@@ -1229,8 +1229,7 @@ void urlSelectSSLBackend()
 		// Could not retrieve ssl backend list - libcurl is probably a slightly older version built with only a single ssl backend
 		return;
 	}
-	// Note: Use CURLSSLBACKEND_DARWINSSL instead of CURLSSLBACKEND_SECURETRANSPORT to support older cURL versions
-	const std::vector<curl_sslbackend> backendPreferencesOrder = {CURLSSLBACKEND_SCHANNEL, CURLSSLBACKEND_DARWINSSL, CURLSSLBACKEND_GNUTLS};
+	const std::vector<curl_sslbackend> backendPreferencesOrder = {CURLSSLBACKEND_SCHANNEL, CURLSSLBACKEND_GNUTLS};
 	std::vector<curl_sslbackend> ignoredBackends;
 #if !defined(USE_OPENSSL_LOCKS_INIT) && !defined(CURL_OPENSSL_DOES_NOT_REQUIRE_LOCKS_INIT)
 	// Did not compile with support for thread-safety / locks for OpenSSL, so ignore it
